ui/app/src/Views/SilenceForm/Updates.elm

```diff
@@ -172,6 +174,55 @@ updateForm msg form =
             in
             { form | matchers = matchers }
 
+        UpdateTimesFromPicker ->
+            let
+                ( startsAt, endsAt, duration ) =
+                    case ( form.pickedStart, form.pickedEnd ) of
+                        ( Just start, Just end ) ->
+                            ( validate timeFromString (initialField (Utils.Date.timeToString start))
+                            , validate (parseEndsAt (Utils.Date.timeToString start)) (initialField (Utils.Date.timeToString end))
+                            , initialField (Utils.Date.durationFormat (Utils.Date.timeDifference start end) |> Maybe.withDefault "")
+                                |> validate Utils.Date.parseDuration
+                            )
+
+                        _ ->
+                            ( form.startsAt, form.endsAt, form.duration )
+            in
+            { form
+                | startsAt = startsAt
+                , endsAt = endsAt
+                , duration = duration
+                , viewDateTimePicker = False
+            }
+
+        OpenDateTimePicker ->
+            let
+                startsAtTime =
+                    case timeFromString form.startsAt.value of
+                        Ok time ->
+                            Just time
+
+                        _ ->
+                            form.pickedStart
+
+                endsAtTime =
+                    case timeFromString form.endsAt.value of
```

Comment:
Pipe to [`Result.toMaybe`](https://package.elm-lang.org/packages/elm/core/latest/Result#toMaybe)